### PR TITLE
Add CloudWatchMeterRegistry

### PR DESCRIPTION
**Pull request by yadavcbala**
*Issue #, if available:*

*Description of changes:*
* Adds CloudWatchRegistry

_Please note that the registry is not currently hooked to data prepper, the current meter registry will still be Prometheus which is already hooked. This PR only adds the CloudWatchRegistry enabling customers who wish to use Cloudwatch to replace Prometheus with CloudWatch_

*Testing after updating the global registry*
Have tested on my dev account and below is the screenshot 

Screen Shot 2021-05-05 at 9 17 07 AM
Screen Shot 2021-05-05 at 9 16 57 AM

-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Provides {@link CloudWatchMeterRegistry} that enables publishing metrics to AWS Cloudwatch. Registry
+ * uses the default aws credentials (i.e. credentials from .aws directory;
+ * refer https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/credentials.html#credentials-file-format).
+ * {@link CloudWatchMeterRegistryProvider} also has a constructor with {@link CloudWatchAsyncClient} that will be used
+ * for communication with Cloudwatch.
+ */
+public class CloudWatchMeterRegistryProvider {
+    private static final String CLOUDWATCH_PROPERTIES = "cloudwatch.properties";
+    private static final Logger LOG = LoggerFactory.getLogger(PrometheusMetricsHandler.class);
```
Comment (chenqi0805) [inline data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java:24]: PrometheusMetricsHandler.class?

**Review by chenqi0805 — COMMENTED** [review]
QUES: just curious. How does the data prepper metrics show in cloudwatch in your test? I did not see anything such as Metrics.addRegistry(getCloudWatchMeterRegistry())

File: data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java (hunk 1 above)
Comment (yadavcbala) [reply data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java:24]: Good catch, will fix it

**Review by chenqi0805 — APPROVED** [review]
LGTM

File: data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * Returns CloudWatchConfig using the properties from {@link #CLOUDWATCH_PROPERTIES}
+     */
+    private CloudWatchConfig createCloudWatchConfig(final String cwPropertiesFile) {
+        CloudWatchConfig cloudWatchConfig = null;
+        try (final InputStream inputStream = Objects.requireNonNull(getClass()
+                .getClassLoader().getResourceAsStream(cwPropertiesFile))) {
+            final Properties cloudwatchProperties = new Properties();
+            cloudwatchProperties.load(inputStream);
+            cloudWatchConfig = cloudwatchProperties::getProperty; //overriding getKey() from CloudWatchConfig
```
Comment (wrijeff) [inline data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java:55]: Just curious - I haven't seen `::` used like this before, what's going on here?

**Review by wrijeff — COMMENTED** [review]
Similar to @chenqi0805's question - how do users enable this in their own deployments? I don't think we have documentation on interchangeable metrics registries but maybe it's time for one?

File: data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java (hunk 2 above)
Comment (yadavcbala) [reply data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/CloudWatchMeterRegistryProvider.java:55]: overriding getKey() from CloudWatchConfig [Added comment as I expected this :-P ]

```
new CloudWatchConfig() {
                @Override
                public String get(String key) {
                    return cloudwatchProperties.getProperty(key);
                }
            }
```

**Comment by yadavcbala** [comment]
> Similar to @chenqi0805's question - how do users enable this in their own deployments? I don't think we have documentation on interchangeable metrics registries but maybe it's time for one?

#575 issue to allow configuring MeterRegistry as part of server configuration, the issue also has the changes required to enable CloudWatchMeterRegistry

**Review by wrijeff — APPROVED** [review]
👍

**Pull request merged**